vivisect/base.py

```diff
@@ -669,19 +677,28 @@ def _initFunction(self, funcva):
     #def _loadImportApi(self, apidict):
         #self._imp_api.update( apidict )
 
+    def getEndian(self):
+        return self.bigend
+
 
 #################################################################
 #
 #  setMeta key callbacks
 #
     def _mcb_Architecture(self, name, value):
+        archid = envi.getArchByName(value)
+        try:
+            # Some of the ENVI archs defined may not architecture modules that 
+            # are still in progress
+            self.setMemArchitecture(archid)
+        except IndexError:
+            raise ArchModDefException(value)
```

Comment:
`raise ArchModDefException(value) from None` to not double up the error messages.

Reply:
```suggestion
            raise ArchModDefException(value) from None
```

Reply:
like this?

Reply:
Yup!